test/ExportQueryExecutionTreesTest.cpp

```diff
@@ -1667,3 +1667,109 @@ TEST(ExportQueryExecutionTrees, convertGeneratorForChunkedTransfer) {
+          if (expected) {
+            EXPECT_THAT(result,
+                        ::testing::Optional(::testing::ResultOf(
+                            [](const auto& literalOrIri) {
+                              return literalOrIri.toStringRepresentation();
+                            },
+                            ::testing::StrEq(*expected))));
+          } else {
+            EXPECT_EQ(result, std::nullopt);
+          }
+        }
+      };
+
+  // Test cases: Each tuple describes one test case.
+  // The first element is the ID of the element to test.
+  // The second element is a list of 2 configurations:
+  // 1. for literals all datatypes except for xsd:string are removed, IRIs
+  // are converted to literals
+  // 2. only literals with `xsd:string` or no datatype are returned
+  std::vector<
+      std::tuple<Id, std::vector<std::tuple<bool, std::optional<std::string>>>>>
+      testCases = {
+          // Case: Literal without datatype
+          {getId("\"something\""),
+           {{false, "\"something\""}, {true, "\"something\""}}},
+
+          // Case: Literal with datatype `xsd:string`
+          {getId("\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"),
+           {{false, "\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"},
+            {true, "\"some\"^^<http://www.w3.org/2001/XMLSchema#string>"}}},
+
+          // Case: Literal with unknown datatype
+          {getId("\"dadudeldu\"^^<http://www.dadudeldu.com/NoSuchDatatype>"),
+           {{false, "\"dadudeldu\""}, {true, std::nullopt}}},
+
+          // Case: IRI
+          {getId("<s>"), {{false, "\"s\""}, {true, std::nullopt}}},
+
+          // Case: datatype `Int`
+          {ad_utility::testing::IntId(1),
+           {{false, "\"1\""}, {true, std::nullopt}}},
+
+          // Case: Undefined ID
+          {ad_utility::testing::UndefId(),
+           {{false, std::nullopt}, {true, std::nullopt}}}};
+
+  for (const auto& [id, cases] : testCases) {
+    checkIdToLiteralOrIri(id, cases);
+  }
+}
+
+TEST(ExportQueryExecutionTrees, IsPlainLiteralOrLiteralWithXsdString) {
+  using Iri = ad_utility::triple_component::Iri;
+  using LiteralOrIri = ad_utility::triple_component::LiteralOrIri;
+  using Literal = ad_utility::triple_component::Literal;
+
+  auto toLiteralOrIri = [](std::string_view content, auto descriptor) {
+    return LiteralOrIri{Literal::literalWithNormalizedContent(
+        asNormalizedStringViewUnsafe(content), descriptor)};
+  };
+
+  auto verify = [](const LiteralOrIri& input, bool expected) {
+    EXPECT_EQ(
+        ExportQueryExecutionTrees::isPlainLiteralOrLiteralWithXsdString(input),
+        expected);
+  };
+
+  verify(toLiteralOrIri("Hallo", std::nullopt), true);
+  verify(toLiteralOrIri(
+             "Hallo",
+             Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>")),
+         true);
+  verify(
+      toLiteralOrIri(
+          "Hallo", Iri::fromIriref("<http://www.unknown.com/NoSuchDatatype>")),
+      false);
```

Comment:
What does this function do, if you pass it an `Iri` (it takes `LiteralOriri` after all).
This case should be tested, probably some exception should be thrown because an assertion fails etc.